File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.cpp
/* A binary tree node

struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};
 */

class Solution {
  public:
    void dfs(Node* root, int level,vector<int>& res) {
        if(root==nullptr) return;
        if(level == res.size()) {
            res.push_back(root->data);
        }
        dfs(root->left,level+1,res);
        dfs(root->right,level+1,res);
    }
    vector<int> leftView(Node *root) {
        // code here
        vector<int> ans;
        int level = 0;
        dfs(root,level,ans);
        return ans;
    }
};

File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.java
/* Structure of Binary Tree Node
class Node {
    int data;
    Node left, right;

    Node(int val) {
        this.data = val;
        this.left = this.right = null;
    }
}*/

class Solution {
    private void solve(Node root,int level, ArrayList<Integer> ans) {
        if(root==null) return;
        if(level==ans.size()) {
            ans.add(root.data);
        }
        
        solve(root.left,level+1,ans);
        solve(root.right,level+1,ans);
    }
    public ArrayList<Integer> leftView(Node root) {
        // code here
        ArrayList<Integer> ans = new ArrayList<>();
        int level = 0;
        solve(root,level,ans);
        return ans;
    }
}